pulser/simulation/simulation.py

```diff
@@ -76,55 +74,155 @@ def __init__(self, sequence: Sequence, sampling_rate: float = 1.0,
         self._tot_duration = max(
             [self._seq._last(ch).tf for ch in self._seq._schedule]
         )
-
         if not (0 < sampling_rate <= 1.0):
-            raise ValueError("The sampling rate (`sampling_rate` = "
-                             f"{sampling_rate}) must be greater than 0 and "
-                             "less than or equal to 1.")
-        if int(self._tot_duration*sampling_rate) < 4:
+            raise ValueError("`sampling_rate` must be positive and "
+                             "not larger than 1.0")
+        if int(self._tot_duration * sampling_rate) < 4:
             raise ValueError("`sampling_rate` is too small, less than 4 data "
                              "points.")
-        self.sampling_rate = sampling_rate
-
+        self._sampling_rate = sampling_rate
         self._qid_index = {qid: i for i, qid in enumerate(self._qdict)}
         self.samples: dict[str, dict[str, dict]] = {
             addr: {basis: {} for basis in ['ground-rydberg', 'digital']}
-            for addr in ['Global', 'Local']
-        }
+            for addr in ['Global', 'Local']}
         self.operators = deepcopy(self.samples)
+        self._collapse_ops: List[qutip.Qobj] = []
+        self._times = self._adapt_to_sampling_rate(
+            np.arange(self._tot_duration, dtype=np.double)/1000)
+        self.evaluation_times = evaluation_times
+        self.config = config if config else SimConfig()
+        self.initial_state = 'all-ground'
+
+    @property
+    def config(self) -> SimConfig:
+        """Property getter for config."""
+        return self._config
+
+    @config.setter
+    def config(self, cfg: SimConfig) -> None:
+        self._config = cfg
+        self._set_param_from_config()
+
+    def show_config(self) -> None:
+        """Shows current configuration."""
+        print(self._config)
+
+    def reset_config(self) -> None:
+        """Resets configuration to default."""
+        self.config = SimConfig()
+        print("Configuration has been set to default.")
+
+    def _set_param_from_config(self) -> None:
+        """Sets all relevant Simulation parameters from its SimConfig.
+
+        Called every time a new configuration is loaded in order to update
+        Simulation parameters.
+        """
+        self._prepare_bad_atoms()
+        # update samples with potential noise settings
+        self._build_hamiltonian_from_seq()
+        if 'dephasing' in self.config.noise:
+            self._init_dephasing()
```

Comment:
This only needs to be done if it hadn't been done already, right?

Reply:
Yes, this goes along with other checks that should ensure `SimConfig`only modifies parameters that did change.

Reply:
I still don't get why this needs to happen every time. In fact, there is no free parameter in `_init_dephasing`, so I would venture in saying it should only happen once overall. 
On that subject, why is the dephasing probability not a free parameter in the 
`SimConfig`?

Reply:
I guess I could add it as a free parameter, it would make sense. Sorry about that, I forgot about checking for new noises before initialising everything. This will be done next commit.